src/Common/ThreadStatus.h

```diff
@@ -204,17 +221,11 @@ class ThreadStatus : public boost::noncopyable
     {
         DetachedFromQuery = 0,  /// We just created thread or it is a background thread
         AttachedToQuery,        /// Thread executes enqueued query
-        Died,                   /// Thread does not exist
     };
 
-    int getCurrentState() const
-    {
-        return thread_state.load(std::memory_order_relaxed);
```

Comment:
Btw I have no idea why `thread_state` is `atomic<int>` but not `atomic<ThreadState>` (and why `ThreadState` is not an enum class). Can we change it as well (or am I missing something?)